packages/gestalt/src/debounce.js

```diff
@@ -5,10 +5,13 @@
  */
 
 // @flow
-export default function debounce(fn: () => void, threshhold: number = 100) {
+export default function debounce(
+  fn: (...args: *) => void,
+  threshhold: number = 100
+) {
   let deferTimer: TimeoutID | null = null;
 
-  const debounced = (...args: any) => {
```

Comment:
Had to change this because there is some confusion in the typing here and worth a revisit. The `*` while deprecated at least lets us bypass the rule for now. Originally `fn` was typed as `fn: () => void` but down below we call it as `fn(...args);` which is contradictory. In actual implementation we never debounce or throttle a function that does take args but I did not want to fully remove the args param as we should keep it generic and revisit later.

Reply:
Can this just be `debounce<T>(fn: T, threshold: number)`?